pkg/kubelet/kubelet.go

```diff
@@ -1971,6 +1978,12 @@ func (kl *Kubelet) syncLoopIteration(updates <-chan kubetypes.PodUpdate, handler
 		// Periodically syncs all the pods and performs cleanup tasks.
 		glog.V(4).Infof("SyncLoop (periodic sync)")
 		handler.HandlePodSyncs(kl.podManager.GetPods())
+	case update := <-kl.livenessManager.Updates():
+		// We only care about failures (signalling container death) here.
+		if update.Result == proberesults.Failure {
+			glog.V(1).Infof("SyncLoop (container unhealthy).")
+			handler.HandlePodSyncs(kl.podManager.GetPods())
```

Comment:
Done. I added the pod object to the update. It's a less elegant API, but I think it's simpler than trying to map the container ID back to the pod.